triples-generator/src/main/scala/io/renku/triplesgenerator/events/consumers/tsprovisioning/triplesuploading/TransformationResultsUploader.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2022 Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.renku.triplesgenerator.events.consumers.tsprovisioning.triplesuploading
+
+import TriplesUploadResult.NonRecoverableFailure
+import cats.MonadThrow
+import cats.data.EitherT
+import cats.effect.Async
+import cats.syntax.all._
+import io.renku.graph.config.{GitLabUrlLoader, RenkuUrlLoader}
+import io.renku.graph.model.entities.{Person, Project}
+import io.renku.graph.model.{GitLabApiUrl, RenkuUrl}
+import io.renku.jsonld.JsonLD.JsonLDEntity
+import io.renku.jsonld.{JsonLD, NamedGraph}
+import io.renku.jsonld.syntax._
+import io.renku.triplesgenerator.events.consumers.{ProcessingRecoverableError, TSVersion}
+import io.renku.triplesstore.{ProjectsConnectionConfig, RenkuConnectionConfig, SparqlQuery, SparqlQueryTimeRecorder}
+import org.typelevel.log4cats.Logger
+
+private trait TransformationResultsUploader[F[_], TS <: TSVersion] {
```

Comment:
Does the `TS <: TSVersion` type parameter needs to be visible to users? Both methods seem not to use it and so I thought to make it an "implementation detail" by using a type member. When using this interface it can be used without knowing the concrete version the uploader is for. I think it is unified in the locator anyways. wdyt?

Reply:
Yes, you are right, that might be better.